google/cloud/testing_util/timer.cc

```diff
@@ -22,19 +22,21 @@ namespace testing_util {
 
 #if GOOGLE_CLOUD_CPP_HAVE_GETRUSAGE
 namespace {
-int rusage_who() {
+int rusage_who(CpuAccounting accounting) {
 #if GOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD
-  return RUSAGE_THREAD;
+  return accounting == CpuAccounting::kPerThread ? RUSAGE_THREAD : RUSAGE_SELF;
 #else
+  (void)accounting;
   return RUSAGE_SELF;
```

Comment:
Opt:  Perhaps @dbolduc's comment would be addressed with something like:
```
switch (accounting) {
  case CpuAccounting::kPerThread:
#if GOOGLE_CLOUD_CPP_HAVE_RUSAGE_THREAD
    return RUSAGE_THREAD;
#else
    return RUSAGE_SELF;
#endif
  case CpuAccounting::kPerProcess:
    return RUSAGE_SELF;
}
```

Reply:
I like that better, with some adjustments to make the compiler(s) happy.